src/rpc/mining.cpp

```diff
@@ -160,12 +160,12 @@ static UniValue generateBlocks(ChainstateManager& chainman, Mining& miner, const
 {
     UniValue blockHashes(UniValue::VARR);
     while (nGenerate > 0 && !chainman.m_interrupt) {
-        std::unique_ptr<CBlockTemplate> pblocktemplate(miner.createNewBlock(coinbase_script));
-        if (!pblocktemplate.get())
+        std::unique_ptr<BlockTemplate> block_template(miner.createNewBlock(coinbase_script));
+        if (!block_template.get())
```

Comment:
In commit "Have createNewBlock return BlockTemplate interface" (6536b345a346670b8733e5372f988452727a2554)

I don't see any way this condition could be true, now or previously. Would suggest changing this to assert(block_template) or CHECK_NONFATAL(block_template)

Same comment also applies to createNewBlock calls below on lines 374 and 817.

The check [here](https://github.com/bitcoin/bitcoin/blob/20ccb30b7af1c30cece2b0579ba88aa101583f07/src/node/miner.cpp#L108-L111) also appears to be dead code. I think it would be good to remove, or replace with an assert, maybe in a separate commit.

Reply:
The only way this _should_ fail is `coinbase_script` is invalid (e.g. too large). But that currently results in a `throw std::runtime_error` inside `BlockAssembler::CreateNewBlock`.

Changed the three lines to `CHECK_NONFATAL`.